numba/cuda/cudadecl.py

```diff
@@ -355,11 +355,50 @@ class Cuda_fp16_binary(ConcreteTemplate):
     return Cuda_fp16_binary
 
 
+def _genfp16_binary_comparison(l_key):
+    @register
+    class Cuda_fp16_cmp(ConcreteTemplate):
+        key = l_key
+
+        cases = [
+            signature(types.b1, types.float16, types.float16)
+        ]
+    return Cuda_fp16_cmp
+
+
+def _genfp16_comparison_operator(l_key):
+    @register_global(l_key)
+    class Cuda_fp16_cmp2(AbstractTemplate):
+        key = l_key
+
+        def generic(self, args, kws):
+            assert not kws
+            if len(args) == 2 and \
+                    (args[0] == types.float16 or args[1] == types.float16):
+                return signature(types.b1, types.float16, types.float16)
+
+    return Cuda_fp16_cmp2
```

Comment:
I don't think there's any need to return the generated class (since we don't need to assign the result of this function to anything).